src/tools/PopulateDB.java

```diff
@@ -8,6 +8,10 @@
 public class PopulateDB {
     private static void makeTables(Connection conn) {
         makeUserTable(conn);
+        makeDriverTable(conn);
+        makeCarTable(conn);
+        makeLocationTable(conn);
```

Comment:
Driver table references Car and Location table (you forgot to reference location table). Don't you need to make those tables first before using them as foreign keys?

Reply:
no need for location table.we just need the location name to see where the driver is not the location coordinates.Hence its better to refrain from using the useless data.